File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/repository/LessonDao.java
package com.mstoyanov.musiclessons.repository;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.mstoyanov.musiclessons.model.Lesson;
import com.mstoyanov.musiclessons.model.LessonWithStudent;

import java.util.List;

@Dao
public interface LessonDao {

    @Query("select lesson.*, student.* " +
            "from lesson " +
            "inner join student on lesson.student_id = student._id " +
            "where lesson.weekday = :weekday")
    List<LessonWithStudent> findAllWithStudentByWeekday(String weekday);

    @Insert
    void insert(Lesson lesson);

    @Update
    void update(Lesson lesson);

    @Delete
    void delete(Lesson lesson);
}

File: music-lessons-2-kotlin/app/src/main/java/com/mstoyanov/musiclessons/repository/LessonDao.kt
package com.mstoyanov.musiclessons.repository

import android.arch.persistence.room.*
import com.mstoyanov.musiclessons.model.Lesson
import com.mstoyanov.musiclessons.model.LessonWithStudent

@Dao
interface LessonDao {

    @Query("select lesson.*, student.* " +
            "from lesson " +
            "inner join student on lesson.student_id = student._id " +
            "where lesson.weekday = :arg0")
    fun findAllWithStudentByWeekday(weekday: String): MutableList<LessonWithStudent>

    @Insert
    fun insert(lesson: Lesson)

    @Update
    fun update(lesson: Lesson)

    @Delete
    fun delete(lesson: Lesson)
}